msgs/include/simple_msgs/numeric_type.hpp

```diff
@@ -140,7 +139,7 @@ class NumericType : public GenericMessage {
    * @brief Addition operator.
    */
   NumericType& operator+=(const NumericType& rhs) {
-    data_ += rhs.data_;
+    data_.exchange(data_.load() + rhs.data_.load());
```

Comment:
Don't want to shit on work that's done already but this would be a good point to ditch the math functionality from your library. Ultimately I think it's not necessary to have these operator overloads (everyone will have their own math types anyways) and it would reduce your codebase, making it easier to maintain. It will not really be a big deal either way but to me it feels cleaner not to include math in a networking lib... Just my two cents. Also, this operation here is not strictly fully threadsafe, and there's no way to fix it unless you put on a full mutex here... :-/

Reply:
I looked up around to find a nicer way to fix this, but now I have to agree with you :\ I'll remove the arithmetical operations and that's it